docs/ClassBasedPlugins.md

```diff
@@ -0,0 +1,131 @@
+
+# Class based plugins
+
+> Experimental
+
+**Note**: It is required to enable the `experimentalDecorators` compiler option to be able to use the typescript decorators.
+
+```ts
+class MyPlugin {
+  constructor (options) {
+    // ...
+  }
+
+  @Get('/')
+  async route (request, reply) {
```

Comment:
I think we should expose the current fastify plugin instance via `this.instance`  or `this.app`. Some decorators will be available there.

Reply:
I'll do that.

Reply:
The current fastify instance is now being exposed.